### PR TITLE
fix kick lottery

### PR DESCRIPTION
**Pull request by zqhxuyuan**
## Description

---

Before we can approve this PR for merge, please make sure that **all** the following items have been checked off:
- [ ] Connected to an issue with discussion and accepted design using zenhub "Connect issue" button below
- [x] Added **one** label out of the `L-` group to this PR
- [ ] Added **one or more** labels from the `A-` and `C-` groups to this PR
- [x] Explicitly labelled `A-calamari` and/or `A-manta` if your changes are meant for/impact either of these (CI depends on it)
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [x] Add `A-integration-test-checks` to run **start-integration-test-checks** (Required)
- [x] Add `A-benchmark-checks` to run **start-benchmark-check** (Required)
- [x] Add `A-unit-test-checks` to run **start-unit-test-checks** (Required)
- [ ] Add `A-congestion-test-checks` to run **start-integration-test-checks** (Optional)

Situational Notes:
- If adding functionality, write unit tests!
- If importing a new pallet, choose a proper module index for it, and allow it in `BaseFilter`. Ensure **every** extrinsic works from front-end. If there's corresponding tool, ensure both work for each other.
- If needed, update our Javascript/Typescript APIs. These APIs are officially used by exchanges or community developers.
- If modifying existing runtime storage items, make sure to implement storage migrations for the runtime and test them with `try-runtime`. This includes migrations inherited from upstream changes, and you can search the diffs for modifications of `#[pallet::storage]` items to check for any.

### CODE REVIEW

File: pallets/pallet-lottery/src/staking/deposit_strategies.rs [hunk 1]
```diff
@@ -40,12 +40,45 @@ pub(super) fn reactivate_bottom_collators<T: Config>(
         let staked = StakedCollators::<T>::get(collator.clone());
         let info = pallet_parachain_staking::Pallet::<T>::candidate_info(collator.clone())
             .expect("is active collator, therefore it has collator info. qed");
+
+        // If collator not exist in delegatorState(PotAccount).delegations, ignore
```
Comment (Dengjianping) [inline pallets/pallet-lottery/src/staking/deposit_strategies.rs:44]: Not sure why use `crate::Pallet::<T>::account_id()` as delegator address.
`delegator_state` is for showing who the delegator is delegating for.

Comment (zqhxuyuan) [reply pallets/pallet-lottery/src/staking/deposit_strategies.rs:44]: user deposit to lottery pool, lottery pool use Pallet account as a delegator to staking module. as many user deposit to the pool, the Pallet account may stake to multiple collator.

Comment (Dengjianping) [reply pallets/pallet-lottery/src/staking/deposit_strategies.rs:44]: Got it

File: pallets/pallet-lottery/Cargo.toml [hunk 2]
```diff
@@ -53,6 +53,7 @@ rand = { workspace = true }
 similar-asserts = { workspace = true }
 sp-staking = { workspace = true, features = ["std"] }
 xcm = { workspace = true, features = ["std"] }
+# env_logger = "0.8.3"
```
Comment (ferrell-code) [inline pallets/pallet-lottery/Cargo.toml]: remove?

File: pallets/pallet-lottery/src/staking/deposit_strategies.rs [hunk 3]
```diff
@@ -40,12 +40,45 @@ pub(super) fn reactivate_bottom_collators<T: Config>(
...
+                }
+            }
+            if is_kick {
+                log::debug!(
+                    "Staked collator:{:?} is kicked off from delegator state, ignore.",
+                    collator
+                );
+                continue;
+            }
+        }
```
Comment (ferrell-code) [inline pallets/pallet-lottery/src/staking/deposit_strategies.rs:62]: Can we remove  from storage `StakedCollators`?

Comment (zqhxuyuan) [reply pallets/pallet-lottery/src/staking/deposit_strategies.rs:62]: let's just ignore the collator for now. maybe need another PR to remove it.

**Review by Dengjianping — APPROVED** [review]

**Review by ferrell-code — APPROVED** [review]

**Pull request merged**